src/adagrams.js

```diff
@@ -1,8 +1,180 @@
 const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
+  drawLetters() {    //wave one, no prams,  return an array of strings
+    const lettersAmount = {   
+      'A': 9,
+      'B': 2,
+      'C': 2,
+      'D': 4,
+      'E': 12,
+      'F': 2,
+      'G': 3,
+      'H': 2,
+      'I': 9,
+      'J': 1,
+      'K': 1,
+      'L': 4,
+      'M': 2,
+      'N': 6,
+      'O': 8,
+      'P': 2,
+      'Q': 1,
+      'R': 6,
+      'S': 4,
+      'T': 6,
+      'U': 4,
+      'V': 2,
+      'W': 2,
+      'X': 1,
+      'Y': 2,
+      'Z': 1,
+    };
+    const tilesArray = [];
+    
+    for (const key in lettersAmount) {
+      for (let countToPush = 0; countToPush < lettersAmount[key]; countToPush++) {
+        tilesArray.push(key); 
+      }  
+    }
+
+    const randArray = [];
+    let i = 0;
+    while (i < 10) {
+      let randNum = (Math.floor((Math.random() * 98) + 1));
+
+      while (randArray.includes(randNum)){
+        randNum = (Math.floor((Math.random() * 98) + 1));
+      }
+      randArray.push(randNum)
+      i++;
+    }
+
+    const hand = [];
+    randArray.forEach(num => hand.push(tilesArray[num]))
+
+    return hand 
+  },
+
+  usesAvailableLetters(input, lettersInHand) {   // wave 2 return true or false if the input letter work with the hand
+    const inputArray = input.split('');
+    let lettersInHandCopy = lettersInHand  //we needed a copy in ruby I guess we need one here too?
```

Comment:
Just in general I suggest making variables `const` unless you need to reassign the variables. 

```suggestion
    const lettersInHandCopy = lettersInHand  //we needed a copy in ruby I guess we need one here too?
```